packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -504,88 +498,38 @@ export function makeVatKeeper(
    * @param {object} entry  The transcript entry to append.
    */
   function addToTranscript(entry) {
-    const oldPos = JSON.parse(getRequired(`${vatID}.t.endPosition`));
+    const oldPos = Number(getRequired(`${vatID}.t.endPosition`));
     const newPos = streamStore.writeStreamItem(
       transcriptStream,
       JSON.stringify(entry),
       oldPos,
     );
-    kvStore.set(`${vatID}.t.endPosition`, `${JSON.stringify(newPos)}`);
+    kvStore.set(`${vatID}.t.endPosition`, `${newPos}`);
   }
 
   /** @returns {StreamPosition} */
   function getTranscriptEndPosition() {
-    return JSON.parse(
+    const endPosition =
       kvStore.get(`${vatID}.t.endPosition`) ||
-        assert.fail('missing endPosition'),
-    );
+      assert.fail('missing endPosition');
+    return Number(endPosition);
   }
 
-  /**
-   * @returns {{ snapshotID: string, startPos: StreamPosition } | undefined}
-   */
-  function getLastSnapshot() {
-    const notation = kvStore.get(`local.${vatID}.lastSnapshot`);
-    if (!notation) {
+  function getSnapshotInfo() {
+    if (snapStore) {
```

Comment:
dunno what your current style preferences are, but I'd be willing to see `return snapStore?.getSnapshotInfo(vatID)` here

Reply:
Dipping our toe in the water of newfangled stuff...